src/test/java/org/amazon/testcases/TC002_GetItemsDetailsWithoutLogin.java

```diff
@@ -0,0 +1,98 @@
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.amazon.base.BaseMethods;
+import org.amazon.library.FileActions;
+import org.amazon.library.PropertyReader;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.Test;
+
+public class TC002_GetItemsDetailsWithoutLogin extends BaseMethods {
+	
+	private static final String IN_STOCK_STATUS = "//div[@id='availability-string']/span";
+	private static final String NET_QUANTITY = "//th[contains(text(), 'Net Quantity')]/following-sibling::td";
+	private static final String BRAND = "//th[contains(text(), 'Brand')]/following-sibling::td";
+	private static final String ITEM_WEIGHT = "//th[contains(text(), 'Item Weight')]/following-sibling::td";
+	private static final String SAVINGS = "//tr[@id='almDetailPagePrice_savings_price']/td[2]/span";
+	private static final String DELIMITER = "#";
+	private static final String MRP = "//tr[@id='almDetailPagePrice_basis_price']/td[2]/span";
+	private static final String SALE_PRICE = "//span[@id='priceblock_ourprice']";
+	private static final String PRODUCT_TITLE = "//span[@id='productTitle']";
+	private final static String ELEMENTS_IN_ONE_PAGE = "//div[@class='a-section a-spacing-micro s-grid-status-badge-container']/a[1]";
+	private FileActions fileactions;
+	
+	@Test
+	public void searchItem() throws InterruptedException, IOException {
+		try {
+			fileactions = new FileActions();
+			home.selectCategory(PropertyReader.configReader("category"));
+			String[] items = PropertyReader.configReader("items").split(",");
+
+			for (String eachItem : items) {
+				getItem(eachItem);
+		    }
+		} 
+		catch (Exception e) {
+			e.printStackTrace();
+			System.out.println("------- Exception occurred -------" + e);
+		}
+	}
+
+	private void getItem(String eachItem) throws IOException {
+		
+		home.enterItemInTextbox(eachItem);
+		home.clickSubmit();
+		
+		WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath(ELEMENTS_IN_ONE_PAGE)));
+		List<WebElement> itemsList = driver.findElements(By.xpath(ELEMENTS_IN_ONE_PAGE));
+		Set<String> mySet = new HashSet<String>();
+		Set<String[]> myFinalSet = new HashSet<String[]>();
+		
+		for (WebElement webElement : itemsList) {
+		    
+			if (mySet.size() >= 10) {
+				break;
+			}
+			else {
+				StringBuilder sb = new StringBuilder();
+				WebDriver newDriver = new ChromeDriver();
+				newDriver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+				String itemLink = webElement.getAttribute("href");
+				String hyperLink = "=HYPERLINK(\"" + itemLink + "\",\"GoToLink\")";
+				
+				if (itemLink.length() < 255) {
```

Comment:
This is wrong. you are skipping the item itself if the link length is greater than 255 - that's not the ask. The ask is that you get the first 10 elements and if any link in those is > 255 characters skip updating the link or update the cell with value `Link too long`

Alternatively you can use some other library to shorten the URL - any standard URL shortener service such as tinyurl or bitly

Reply:
Done.